perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -6058,6 +6087,43 @@ public void checkAttributeDependencies(PerunSession sess, RichAttribute richAttr
 	}
 
 	/**
+	 * Checks if the attributes represent empty values. If so, converts them into null.
+	 *
+	 * @param attributes attributes to be checked if are null
+	 */
+	public void convertEmptyAttrValueToNull(List<Attribute> attributes) {
+		attributes.forEach(this::convertEmptyAttrValueToNull);
+	}
+
+	/**
+	 * Checks if the attribute represents empty value. If so, converts it into null.
+	 *
+	 * @param attribute attribute to be checked if is null
+	 */
+	public void convertEmptyAttrValueToNull(Attribute attribute) {
```

Comment:
Please add usage of this method to 
```java
public Attribute mergeAttributeValue(PerunSession sess, User user, Attribute attribute)
```
in ``AttributeManagerBlImpl`` where you should use input attribute object and call it before line

```java
if(attribute.getValue() == null) return storedAttribute;
```
so we can be sure, that empty attribute is really null.